guest/klog/klog.go

```diff
@@ -0,0 +1,79 @@
+/*
+   Copyright 2023 The Kubernetes Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package klog
+
+import (
+	"runtime"
+
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/mem"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/klog/api"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/klog/internal"
+)
+
+// Get can be called at any time, to get a logger that writes to the
+// WebAssembly host.
+//
+// For example:
+//
+//	func main() {
+//		klog := klog.Get()
+//		klog.Info("hello", "world")
+//	}
+func Get() api.Klog {
+	return instance
+}
+
+var instance api.Klog = &internal.Klog{
+	Severity: severity(),
+	LogFn:    logFn,
+}
+
+func logFn(severity internal.Severity, msg []byte) {
+	ptr, size := mem.BytesToPtr(msg)
+	log(severity, ptr, size)
+	runtime.KeepAlive(msg) // keep msg alive until ptr is no longer needed.
+}
+
+// Info is a convenience that calls the same method documented on api.Klog.
+//
+// Note: Code that uses can be unit tested in normal Go, but cannot be unit
```

Comment:
I added these for parity with the normal klog package, with warnings for folks who test with tinygo. This is an acknowledgement that few will, and so these make code easier for those people.